olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -46,31 +46,14 @@ VersionedLayerClientImpl::VersionedLayerClientImpl(
     client::OlpClientSettings settings)
     : catalog_(std::move(catalog)),
       layer_id_(std::move(layer_id)),
-      settings_(
-          std::make_shared<client::OlpClientSettings>(std::move(settings))),
+      settings_(std::move(settings)),
       pending_requests_(std::make_shared<client::PendingRequests>()) {
-  if (!settings_->cache) {
-    settings_->cache = client::OlpClientSettingsFactory::CreateDefaultCache({});
+  if (!settings_.cache) {
+    settings_.cache = client::OlpClientSettingsFactory::CreateDefaultCache({});
   }
   // to avoid capturing task scheduler inside a task, we need a copy of settings
   // without the scheduler
-  task_scheduler_ = std::move(settings_->task_scheduler);
-
-  auto api_repo = std::make_shared<repository::ApiRepository>(
-      catalog_, settings_, settings_->cache);
-
-  auto catalog_repo = std::make_shared<repository::CatalogRepository>(
-      catalog_, api_repo, settings_->cache);
-
-  partition_repo_ = std::make_shared<repository::PartitionsRepository>(
-      catalog_, layer_id_, api_repo, catalog_repo, settings_->cache);
-
-  auto data_repo = std::make_shared<repository::DataRepository>(
-      catalog_, layer_id_, api_repo, catalog_repo, partition_repo_,
-      settings_->cache);
-
-  prefetch_repo_ = std::make_shared<repository::PrefetchTilesRepository>(
-      catalog_, layer_id_, settings_);
+  task_scheduler_ = std::move(settings_.task_scheduler);
```

Comment:
Is this still the case? Now that we fully use TaskContext this shouldn't be the case anymore.